### PR TITLE
Fix: Check extra scope in no-use-before-define (fixes #2372)

### LINKED ISSUE #2372: "no-use-before-define" not catching any violations when env=node
Hey all, I just realized that eslint wasn't picking up a "no-use-before-define" rule violation in my codebase. I created a small test file based and eslint was still unable to catch it:

```js
// test.js
console.log(a);
var a = 'abcdefg';
```

```
$ eslint -v
v0.19.0
$ eslint --reset --rule 'no-use-before-define: 2' test.js
$ echo $?
0
```

It looks like this rule is heavily tested in eslint itself, any idea why this wouldn't be catching? All other rules seem to be working fine.

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: lib/rules/no-use-before-define.js [hunk 1]
```diff
@@ -58,10 +70,30 @@ module.exports = function(context) {
...
+        "Program": function() {
+            var scope = context.getScope();
+            findVariables(scope);
+
+            // both Node.js and Modules have an extra scope
+            if (context.ecmaFeatures.globalReturn || context.ecmaFeatures.modules) {
+                findVariables(scope.childScopes[0]);
+            }
+        },
+        "FunctionExpression": findVariablesInNode,
```
Comment (ilyavolodin) [inline lib/rules/no-use-before-define.js]: I think the name is a bit confusing. What about finding variables in browser? (I know it does both, just making a point). Also, wouldn't it be easier to modify findVariables to just do `scope = scope || context.getScope()`.

Comment (nzakas) [inline lib/rules/no-use-before-define.js]: lol, point taken

Comment (nzakas) [inline lib/rules/no-use-before-define.js]: Can't actually do `scope = scope || context.getScope()` since the AST node is passed as an argument.

Comment (ilyavolodin) [inline lib/rules/no-use-before-define.js]: Right forgot about that. I guess it's fine then. The name is still a bit confusing though.

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**